lib/experimental/Navigation/Header/PageHeader/index.tsx

```diff
@@ -74,21 +83,31 @@ export function PageHeader({
           )}
         </div>
       </div>
-      {actions && (
-        <div className="flex items-center gap-2">
-          {actions.map((action, index) => (
-            <Button
-              hideLabel
-              round
-              key={index}
-              variant="outline"
-              onClick={action.onClick}
-              label={action.label}
-              icon={action.icon}
-            />
-          ))}
-        </div>
-      )}
+      <div className="flex items-center">
+        {!hasNavigation && hasStatus && (
+          <div className="pe-3">
+            <StatusTag text={statusTag.text} variant={statusTag.variant} />
+          </div>
+        )}
+        {hasStatus && hasActions && (
+          <div className="right-0 h-4 w-px bg-f1-border-secondary"></div>
+        )}
+        {hasActions && (
+          <div className="items-right flex gap-2 ps-3">
+            {actions.map((action, index) => (
+              <Button
+                hideLabel
+                round
+                key={index}
+                variant="outline"
+                onClick={action.onClick}
+                label={action.label}
+                icon={action.icon}
+              />
+            ))}
+          </div>
```

Comment:
We don't have to do it now, but I can't help but think this could be expressed with something like:

```
{
  [
    hasStatus ? <div className ...> : undefined,
    hasActions ? actions.map(() => ...) : undefined,
  ].filter(Boolean).join(<div className={} ... />)
}
```